Add sub-group: uncaught exception

JabRef 5.3--2021-07-15--cb8fb55
Linux 4.19.0-17-amd64 amd64 
Java 16.0.1 
JavaFX 16+8

- [X] **Mandatory**: I have tested the latest development version from http://builds.jabref.org/main and the problem persists

When adding a sub-group, an uncaught exception is raised, and no sub-group is created.

Steps to reproduce the behavior:
1. In the group panel, right-click on a group name and select "Add subgroup"

Note 1: same behavior when clicking on the "Add group" button located at the bottom of the group panel.
Note 2: it worked in JabRef 5.3--2021-07-05--50c96a2 (the release version) and in JabRef 5.3--2021-07-07--2fe7f31
Note 3: Since version 5.3 is now released, shouldn't the development versions be called JabRef **5.4**--... ? (accounted for in #7913)

<!-- If applicable, add excerpt of the bibliography file, screenshot, and excerpt of log (available in the error console) -->

<details>
  <summary>Exception details</summary>
Uncaught exception occured in Thread[JavaFX Application Thread,5,main]  

Cannot invoke "java.util.Enum.Set.contains(Object)" because the return value of "javafx.beans.property.ObjectProperty.getValue()" is null

java.lang.NullPointerException: Cannot invoke "java.util.EnumSet.contains(Object)" because the return value of "javafx.beans.property.ObjectProperty.getValue()" is null
	at org.jabref@5.3.19/org.jabref.gui.groups.GroupDialogViewModel.lambda$setupValidation$5(Unknown Source)
	at org.jabref.merged.module@5.3.19/de.saxsys.mvvmfx.utils.validation.FunctionBasedValidator.lambda$new$2(Unknown Source)
	at org.jabref.merged.module@5.3.19/de.saxsys.mvvmfx.utils.validation.FunctionBasedValidator.validate(Unknown Source)
	at org.jabref.merged.module@5.3.19/de.saxsys.mvvmfx.utils.validation.FunctionBasedValidator.<init>(Unknown Source)
	at org.jabref@5.3.19/org.jabref.gui.groups.GroupDialogViewModel.setupValidation(Unknown Source)
	at org.jabref@5.3.19/org.jabref.gui.groups.GroupDialogViewModel.<init>(Unknown Source)
	at org.jabref@5.3.19/org.jabref.gui.groups.GroupDialogView.<init>(Unknown Source)
	at org.jabref@5.3.19/org.jabref.gui.groups.GroupTreeViewModel.lambda$addNewSubgroup$7(Unknown Source)
	at java.base/java.util.Optional.ifPresent(Unknown Source)
	at org.jabref@5.3.19/org.jabref.gui.groups.GroupTreeViewModel.addNewSubgroup(Unknown Source)
	at org.jabref@5.3.19/org.jabref.gui.groups.GroupTreeView.lambda$createContextMenuForGroup$23(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/javafx.event.Event.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/javafx.scene.control.MenuItem.fire(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.scene.control.ContextMenuContent$MenuItemContainer.doSelect(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.scene.control.ContextMenuContent$MenuItemContainer.lambda$createChildren$12(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/javafx.event.Event.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/javafx.scene.Scene$MouseHandler.process(Unknown Source)
	at org.jabref.merged.module@5.3.19/javafx.scene.Scene.processMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/javafx.scene.Scene$ScenePeerListener.mouseEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.glass.ui.View.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.glass.ui.View.notifyMouse(Unknown Source)
	at org.jabref.merged.module@5.3.19/com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at org.jabref.merged.module@5.3.19/com.sun.glass.ui.gtk.GtkApplication.lambda$runLoop$11(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
</details>

I encounter the same problem in Windows 7

JabRef 5.4--2021-07-19--5624b85
Windows 7 6.1 amd64 
Java 16.0.1 
JavaFX 16+8

<details open>
<summary>Exception details</summary>

`java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at org.jabref.merged.module@5.4.29/javafx.fxml.FXMLLoader$MethodHandler.invoke(Unknown Source)
	at org.jabref.merged.module@5.4.29/javafx.fxml.FXMLLoader$ControllerMethodEventHandler.handle(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/javafx.event.Event.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/javafx.scene.Node.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/javafx.scene.control.Button.fire(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.scene.control.behavior.ButtonBehavior.mouseReleased(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.scene.control.inputmap.InputMap.handle(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/javafx.event.Event.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/javafx.scene.Scene$MouseHandler.process(Unknown Source)
	at org.jabref.merged.module@5.4.29/javafx.scene.Scene.processMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/javafx.scene.Scene$ScenePeerListener.mouseEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.glass.ui.View.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.glass.ui.View.notifyMouse(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at org.jabref.merged.module@5.4.29/com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at com.sun.javafx.reflect.Trampoline.invoke(Unknown Source)
	at jdk.internal.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.reflect.MethodUtil.invoke(Unknown Source)
	at org.jabref.merged.module@5.4.29/com.sun.javafx.fxml.MethodHelper.invoke(Unknown Source)
	... 55 more
Caused by: java.lang.NullPointerException: Cannot invoke "java.util.EnumSet.contains(Object)" because the return value of "javafx.beans.property.ObjectProperty.getValue()" is null
	at org.jabref@5.4.29/org.jabref.gui.groups.GroupDialogViewModel.lambda$setupValidation$5(Unknown Source)
	at org.jabref.merged.module@5.4.29/de.saxsys.mvvmfx.utils.validation.FunctionBasedValidator.lambda$new$2(Unknown Source)
	at org.jabref.merged.module@5.4.29/de.saxsys.mvvmfx.utils.validation.FunctionBasedValidator.validate(Unknown Source)
	at org.jabref.merged.module@5.4.29/de.saxsys.mvvmfx.utils.validation.FunctionBasedValidator.<init>(Unknown Source)
	at org.jabref@5.4.29/org.jabref.gui.groups.GroupDialogViewModel.setupValidation(Unknown Source)
	at org.jabref@5.4.29/org.jabref.gui.groups.GroupDialogViewModel.<init>(Unknown Source)
	at org.jabref@5.4.29/org.jabref.gui.groups.GroupDialogView.<init>(Unknown Source)
	at org.jabref@5.4.29/org.jabref.gui.groups.GroupTreeViewModel.lambda$addNewSubgroup$7(Unknown Source)
	at java.base/java.util.Optional.ifPresent(Unknown Source)
	at org.jabref@5.4.29/org.jabref.gui.groups.GroupTreeViewModel.addNewSubgroup(Unknown Source)
	at org.jabref@5.4.29/org.jabref.gui.groups.GroupTreeViewModel.addNewGroupToRoot(Unknown Source)
	at org.jabref@5.4.29/org.jabref.gui.groups.GroupTreeView.addNewGroup(Unknown Source)
	... 65 more`
</details>

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -21,6 +21,7 @@ Note that this project **does not** adhere to [Semantic Versioning](http://semve
 ### Fixed
 
 - We fixed an issue when checking for a new version when JabRef is used behind a corporate proxy. [#7884](https://github.com/JabRef/jabref/issues/7884)
+- We fixed an issue where it was impossible to add or modify groups. [#7930](https://github.com/JabRef/jabref/pull/793://github.com/JabRef/jabref/pull/7930)
 
 ### Removed
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -21,7 +21,7 @@ Note that this project **does not** adhere to [Semantic Versioning](http://semve
 ### Fixed
 
 - We fixed an issue when checking for a new version when JabRef is used behind a corporate proxy. [#7884](https://github.com/JabRef/jabref/issues/7884)
-- We fixed an issue where it was impossible to add or modify groups. [#7930](https://github.com/JabRef/jabref/pull/793://github.com/JabRef/jabref/pull/7930)
+- We fixed an issue where it was impossible to add or modify groups. [#7912](https://github.com/JabRef/jabref/pull/793://github.com/JabRef/jabref/pull/7921)
 
 ### Removed
 

--- a/src/main/java/org/jabref/gui/groups/GroupDialogViewModel.java
+++ b/src/main/java/org/jabref/gui/groups/GroupDialogViewModel.java
@@ -83,7 +83,7 @@ public class GroupDialogViewModel {
     private final BooleanProperty keywordGroupRegexProperty = new SimpleBooleanProperty();
 
     private final StringProperty searchGroupSearchTermProperty = new SimpleStringProperty("");
-    private final ObjectProperty<EnumSet<SearchFlags>> searchFlagsProperty = new SimpleObjectProperty<>();
+    private final ObjectProperty<EnumSet<SearchFlags>> searchFlagsProperty = new SimpleObjectProperty(EnumSet.noneOf(SearchFlags.class));
 
     private final BooleanProperty autoGroupKeywordsOptionProperty = new SimpleBooleanProperty();
     private final StringProperty autoGroupKeywordsFieldProperty = new SimpleStringProperty("");

